### PR TITLE
Require windows TFM if MyType requires that

### LINKED ISSUE #434: <MyType> in VB generates calls to unsupported methods
If a VB project has something similar to:

```xml
<PropertyGroup>
<MyType>Windows</MyType>
</PropertyGroup>
```

Then the build will fail with:

```
vbc : error BC30002: Type 'Global.Microsoft.VisualBasic.Devices.Computer' is not defined.
```

### PR DESCRIPTION
**Pull request by twsouthwick**
Fixes #434

### CODE REVIEW

**Review by mjrousos — APPROVED** [review]
LGTM

File: src/components/Microsoft.DotNet.UpgradeAssistant/UpgraderExtensions.cs [hunk 1]
```diff
@@ -38,5 +42,10 @@ private static void AddTargetFrameworkSelectors(this IServiceCollection services
             services.AddTransient<ITargetFrameworkSelectorFilter, WindowsSdkTargetFrameworkSelectorFilter>();
             services.AddTransient<ITargetFrameworkSelectorFilter, ExecutableTargetFrameworkSelectorFilter>();
         }
+
+        private static void AddVisualBasic(this IServiceCollection services)
+        {
+            services.AddTransient<ITargetFrameworkSelectorFilter, MyNamespaceTargetFrameworkSelectorFilter>();
```
[the remark this answers is not in the document]
Comment (twsouthwick) [reply src/components/Microsoft.DotNet.UpgradeAssistant/UpgraderExtensions.cs:48]: Good question - do we want to add "selectors" or add support for feature areas? I may take a look at how we're structuring that and consolidate them into feature areas so we understand what is needed for each feature area

**Review by KSchlobohm — APPROVED** [review]
Seems like a reasonable experiment. I'd like to see what customers think about this as an option.

File: src/components/Microsoft.DotNet.UpgradeAssistant/UpgraderExtensions.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (KSchlobohm) [reply src/components/Microsoft.DotNet.UpgradeAssistant/UpgraderExtensions.cs]: I prefer feature areas, that would be more descriptive. Until then, this increases the concept count and would require that I know both.

[the remark this answers is not in the document]
Comment (twsouthwick) [reply src/components/Microsoft.DotNet.UpgradeAssistant/UpgraderExtensions.cs]: I'll create an issue to separate things into feature areas, but will stay consistent for now.

**Pull request merged**